test/run-drun/idx.mo

```diff
@@ -0,0 +1,27 @@
+actor a {
+    type O1 = { field : Int }; // "\ba\94\93\00"
+    type O2 = { a: Int; field : Int }; // "a\00\00\00\ba\94\93\00"
+    type O3 = { a: Int; field : Int; other : Int }; // "a\00\00\00\ba\94\93\00\d0fv6"
+    type O4 = { a: Int; foo : Int; field : Int; other : Int }; // "a\00\00\00\06\c7M\00\ba\94\93\00\d0fv6"
+
+    func go1(o : O1) : async () = async inner o;
+    func go2(o : O2) : async () = async inner o;
+    func go3(o : O3) : async () = async inner o;
+    public shared func go4(o : O4) : async () { assert o.a == 25; assert o.field == 42; assert o.other == 83; assert o.foo == 8; inner o };
+
+    func inner(o : O1) { assert o.field == 42 };
+
+    public func go() : async () {
+        await go1({ field = 42 }); // field: 9671866
+        await go2({ a = 25; field = 42 });
+        await go3({ a = 25; field = 42; other = 83 });
+        await go4({ a = 25; foo = 8; field = 42; other = 83 })
+    }
```

Comment:
Can you add a test that projects a field from a value of type None (generated with loop `{}`). That would test code generated for impossible values.

Reply:
Isn't that dead code, so that codegen won't insert the field access, but `unreachable` instead?

Reply:
Uhm, I added
``` swift
debug {
    let l : O4 = loop {};
    go4(l)
}
```
at the end, which compiles, but doesn't terminate!

Reply:
Is this what you mean?

Reply:
No, I meant  something more direct like  `(loop {}).foo`, where the unreachable code is directly project from. These kind of examples have bitten us before. I'm sure it'll be fine but worth checking.

Reply:
If the non-termination is a problem, just put it behind a lambda to test code gen doesn't fail, without actually invoking it in the test.

Reply:
PTAL: #2727